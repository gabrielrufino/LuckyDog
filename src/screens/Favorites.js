import React, {useCallback, useState, useEffect} from 'react';
import {
  ScrollView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';

import Button from '../components/Button';
import DogImage from '../components/DogImage';
import FavoriteRepository from '../repositories/Favorite';
import Header from '../components/Header';
import useBreeds from '../hooks/useBreeds';

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#FFFFFF',
    flex: 1,
  },
  scrollView: {
    marginTop: 10,
    justifyContent: 'center',
  },
  dogImageContainer: {
    height: '95%',
    paddingHorizontal: 10,
    width: 320,
  },
  emptyWarning: {
    textAlign: 'center',
  },
});

export default function Favorites({navigation}) {
  const [favorites, setFavorites] = useState([]);
  const {breeds} = useBreeds();

  useEffect(
    function () {
      console.log(breeds);
    },
    [breeds],
  );

  useEffect(function () {
    FavoriteRepository.getAll().then(function (data) {
      setFavorites(data);
    });
  }, []);

  const onExpand = useCallback(
    function (favorite) {
      navigation.navigate('Dog', {image: favorite.uri});
    },
    [navigation],
  );

  return (
    <View style={styles.container}>
      <Header title="Favoritos" />

      {!favorites.length && (
        <Text style={styles.emptyWarning}>
          Você ainda não tem dogs favoritos
        </Text>
      )}

      <ScrollView contentContainerStyle={styles.scrollView} horizontal>
        {favorites.map((favorite) => (
          <TouchableOpacity
            key={favorite.uri}
            style={styles.dogImageContainer}
            onPress={() => onExpand(favorite)}>
            <DogImage source={{uri: favorite.uri}} />
            <Button label="Expandir" flat onPress={() => onExpand(favorite)} />
          </TouchableOpacity>
        ))}
      </ScrollView>
    </View>
  );
}
