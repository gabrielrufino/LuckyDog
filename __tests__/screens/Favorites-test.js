jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(() => ({
    goBack: jest.fn(),
  })),
}));

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Favorites from '../../src/screens/Favorites';

describe('screens/Favorites', () => {
  it('Should renders the favorites screen correctly', () => {
    const tree = renderer.create(<Favorites />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
