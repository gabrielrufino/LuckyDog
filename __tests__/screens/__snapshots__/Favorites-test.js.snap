// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screens/Favorites Should renders the favorites screen correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "padding": 15,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#C8AB83",
              "fontSize": 25,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome5Free-Solid",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "fontWeight": "900",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 5,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#C8AB83",
          "fontSize": 25,
        }
      }
    >
      Favoritos
    </Text>
  </View>
  <Text
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Você ainda não tem dogs favoritos
  </Text>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "justifyContent": "center",
        "marginTop": 10,
      }
    }
    horizontal={true}
  >
    <View />
  </RCTScrollView>
</View>
`;
